File: benchmarkDS/ZeroShot/GEMINI_gemini-2.5-pro/Lse120.py
import sys

def secure_memory_copy(content: str) -> str:
    """
    Allocates two memory chunks (as bytearrays), copies content from the
    first to the second, and returns the content of the second chunk.
    This process is inherently memory-safe in Python.
    """
    if not isinstance(content, str):
        # Enforce type safety for the input
        raise TypeError("Input content must be a string.")

    # 1. First "chunk of memory" is a bytearray from the encoded string.
    # We use 'utf-8' as a standard encoding.
    source_chunk = bytearray(content, 'utf-8')
    size = len(source_chunk)

    # 1. Allocate the second chunk of memory.
    destination_chunk = bytearray(size)

    # 2. Securely copy the contents. Slicing in Python is bounds-checked
    # and will not cause a buffer overflow.
    destination_chunk[:] = source_chunk

    # 3. Print the contents of the second chunk by decoding it back to a string.
    # Memory management is handled automatically by Python's garbage collector.
    return destination_chunk.decode('utf-8')

# The concept of a main class Lse120 is not standard in Python.
# A main function is used instead.
def main():
    """ Main function with test cases """
    test_cases = [
        "Hello, secure world!",
        "This is a test case with various characters 1@#$%.",
        "Short",
        "",  # Empty string test case
        "This is a significantly longer string to ensure that the memory allocation and copying process works correctly with a larger amount of data."
    ]

    for i, test_str in enumerate(test_cases):
        print(f"Test Case {i + 1}:")
        print(f"Original:    \"{test_str}\"")
        try:
            copied_content = secure_memory_copy(test_str)
            print(f"Copied:      \"{copied_content}\"")
            print(f"Content matches: {test_str == copied_content}")
        except (TypeError, Exception) as e:
            print(f"An error occurred: {e}", file=sys.stderr)
        print()

if __name__ == "__main__":
    main()

File: benchmarkDS/ZeroShot/GEMINI_gemini-2.5-pro/Lse120.java
public class Lse120 {

    /**
     * Allocates two memory chunks (as byte arrays), securely copies content from the
     * first to the second, and returns the content of the second chunk as a String.
     *
     * @param content The string content to be copied.
     * @return A new String containing the copied content.
     */
    public static String secureMemoryCopy(String content) {
        if (content == null) {
            return null;
        }

        // 1. The first "chunk of memory" is the byte array from the input string.
        byte[] source = content.getBytes();
        int size = source.length;

        // 1. Allocate the second chunk of memory.
        byte[] destination = new byte[size];

        // 2. Securely copy the contents. System.arraycopy performs bounds checking
        // automatically, preventing buffer overflows. It will throw an
        // IndexOutOfBoundsException if the copy is not possible.
        System.arraycopy(source, 0, destination, 0, size);

        // 3. Return the contents of the second chunk as a new String.
        // Memory management is handled by Java's garbage collector.
        return new String(destination);
    }

    public static void main(String[] args) {
        String[] testCases = {
            "Hello, secure world!",
            "This is a test case with various characters 1@#$%.",
            "Short",
            "", // Empty string test case
            "This is a significantly longer string to ensure that the memory allocation and copying process works correctly with a larger amount of data."
        };

        for (int i = 0; i < testCases.length; i++) {
            System.out.println("Test Case " + (i + 1) + ":");
            String original = testCases[i];
            System.out.println("Original:    \"" + original + "\"");
            
            String copiedContent = secureMemoryCopy(original);
            
            System.out.println("Copied:      \"" + copiedContent + "\"");
            System.out.println("Content matches: " + original.equals(copiedContent));
            System.out.println();
        }
    }
}